spi/src/main/java/org/eclipse/dataspaceconnector/spi/types/domain/asset/Asset.java

```diff
@@ -48,91 +51,73 @@ public String getVersion() {
         return version;
     }
 
-    public String getFileName() {
```

Comment:
see the comment above.

Reply:
This goes back to the previous discussions we've had on the architectural vision we have for the EDC. We'd like it to be a general data-sharing platform, supporting IDS and other protocols provided through extensions.

This requires (among other things) the platform to be architected so that it can convey extensible types. One analogy that may help clarify this statement is that we are building software akin to messaging middleware. Those systems, by definition, have a concept of a "message," which nearly always contains a header and payload; they don't know anything about the semantics of the data they carry. 

The *core* EDC modules should be similar to this. Extensions "above" and "below" the core may understand additional qualities about the extensible types. To make this a little more concrete, the issue with the fileName can be handled with this architectural principle in mind by treating Asset as an extensible type (which it should be). I would not expect the base Asset type to include the union of all properties mandated by all protocols EDC supports. Because IDS defines it (IMO a bug in the spec that we can easily change), it does not mean we should include it on the base Asset class. 

There are two ways of handling extensible types, and I think both are valid. One is to use extensible properties (e.g. a map). The IDS InfoModel does this as well as many other information models. That the IDS folks thought of adding those properties is a good thing. Another way of doing this is to have lower or higher-level services such as the IDS modules and AssetIndex implementations subclass Asset combined with Jackson's ability to handle polymorphic (de)/serialization. That's why we have the TypeManager in the EDC (which also handles this in the Jersey JAX-RS container.

I prefer the loosey-goosey extensible properties approach because it avoids the proliferation of modules containing only types and plays nicely with schemaless and Graph databases.  However, I think that discussion is separate from this PR as the latter is concerned with the core and not how extensible types are handled in other architectural layers.

Does that make sense?

Reply:
@jimmarino Yea, I think you are right. My thoughts came from a wrong assumption about the asset index and how it would be implemented :) Paul was able to convince me in a spontaneous meeting this afternoon.